File: aizu/tyama_aizu0168.pl
#!/usr/bin/perl
use strict;
use warnings;
my @a=(1,1,2);
for(my $i=3;$i<31;$i++){push(@a,$a[$i-1]+$a[$i-2]+$a[$i-3]);}
for(;my $n=int(<>);){print int(($a[$n]+3649)/3650)."\n";}

File: aizu/tyama_aizu0168.py
#!/usr/bin/python
import sys
if sys.version_info[0]>=3: raw_input=input
a=[1,1,2]
for i in range(3,31): a.append(a[i-1]+a[i-2]+a[i-3])
while True:
	n=int(raw_input())
	if n==0: break
	print((a[n]+3649)//3650)